### ISSUE TITLE
Add support for declaring named constants (scalars, vectors, matrices)

### ISSUE THREAD

**Issue by tohrnii** [0xPolygonMiden/air-script#60]
We should add support for constants. The syntax was defined by @bobbinth in the original Constraints Description Language discussion [here](https://github.com/maticnetwork/miden/discussions/254#discussioncomment-3284984). Taken directly from @bobbinth's original post:

Constants could be of the following three types:
- scalars
- vectors
- matrices

The constants should be declared in a separate section.
 
Declaring constants could be done like so:
```
constants:
  a: 123 // a scalar constant
  b: [1, 2, 3] // a vector constant
  c: [
    [1, 2, 3],
    [4, 5, 6],
  ] // a matrix constant
```

Referring to vector constants inside expressions could be done with index notation:
```
let x = b[1]
let y = c[0][2]
```

Something like this could also be possible:
```
let x = c[1][1..3] // this sets x to [5, 6]
```

Other open suggestions by @grjte and @bobbinth to easily distinguish between trace columns and constants:
- Use CAPITALS for constant names. If we do this, we should decide if we want to show an error to the user or a warning to use correct convention.
- Another option is to prefix trace columns with a some special symbol.

This task involves making changes to the parser, IR and codegen.

- [ ] parser
- [ ] IR
- [ ] codegen

**Comment by tohrnii** [comment]
Add validation for column number mismatch for rows to the IR. For eg, the following should throw an error:
```
A: [[1, 2, 3], [4, 5]]
```

**Comment by tohrnii** [comment]
Closed by #63, #94 and #95.
